scripts/upgrade.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+set -eEuo pipefail
+
+ROOT_DIR="$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")/.." &>/dev/null && pwd)"
+source "${ROOT_DIR}"/scripts/utils.sh
+
+verify_required_commands
+
+echo -e "\n\033[1mWhat kind of node type would you like to run: \033[0m"
+select role_value in rpc forger; do
+  if [ -n "${role_value}" ]; then
+    echo -e "\nYou have selected: \033[1m${role_value}\033[0m"
+    break
+  else
+    echo -e "\n\033[1mInvalid selection. Please type 1, or 2.\033[0m\n"
+  fi
+done
+
+echo -e "\n\033[1mWhat network would you like to setup 'eon' (mainnet) or 'gobi' (testnet): \033[0m"
+select network_value in eon gobi; do
+  if [ -n "${network_value}" ]; then
+    echo -e "\nYou have selected: \033[1m${network_value}\033[0m"
+    break
+  else
+    echo -e "\n\033[1mInvalid selection. Please type 1, or 2.\033[0m\n"
+  fi
+done
+
+DEPLOYMENT_DIR="${ROOT_DIR}/deployments/${role_value}/${network_value}"
+ENV_FILE_TEMPLATE="${ROOT_DIR}/env/.env.${role_value}.${network_value}.template"
+ENV_FILE="${DEPLOYMENT_DIR}/.env"
+
+# Create a backup of .env file called .env.bk
+echo -e "\n\033[1m=== Creating a backup of the .env file ===\033[0m"
+cp "${ENV_FILE}" "${ENV_FILE}.bk"
+
+# Define the auto update variables
+auto_update_vars=(
+  "EVMAPP_TAG"
+  "ZEND_TAG"
+  "ZEN_SEED_TAR_GZ_URL"
+)
+
+conditional_update_vars=(
+  "ZEN_OPTS"
+  "SCNODE_FORGER_MAXCONNECTIONS"
+  "SCNODE_NET_MAX_IN_CONNECTIONS"
+  "SCNODE_NET_MAX_OUT_CONNECTIONS"
+  "SCNODE_LOG_FILE_LEVEL"
+)
+
+# Read the .env.template file line by line, skip blanc lines and comments, store each of the other lines in an array
+echo -e "\n\033[1m=== Reading .env.template file ===\033[0m"
+while IFS= read -r line; do
+  [ -z "$line" ] && continue
+  [ "${line:0:1}" = "#" ] && continue
+  env_template_lines+=("$line")
+done <"$ENV_FILE_TEMPLATE"
```

Comment:
I am a fan of {} around vars but it is up to you :). If you decided to fix it there are more occurrences of this